main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeTravelSpeedValidator.java

```diff
@@ -0,0 +1,360 @@
+      // endIdx, computed as the sum of straight-line distances between consecutive stops.
+      double distanceToEndIdx = 0;
+      for (int startIdx = endIdx - 1; startIdx >= 0; --startIdx) {
+        final GtfsStopTime startStopTime = stopTimes.get(startIdx);
+        distanceToEndIdx += distancesKm[startIdx];
+        if (!startStopTime.hasDepartureTime()) {
+          continue;
+        }
+        final double speedKph =
+            getSpeedKphBetweenStops(distanceToEndIdx, startStopTime, endStopTime);
+        if (speedKph <= maxSpeedInKph) {
+          continue;
+        }
+        GtfsStop startStop = stopTable.byStopId(startStopTime.stopId());
+        // Give a notice if they are too far apart.
+        if (distanceToEndIdx > MAX_DISTANCE_OVER_MAX_SPEED_IN_KMS) {
+          // Issue one notice per each trip.
+          for (List<GtfsStopTime> stopsOnTrip : trips) {
+            noticeContainer.addValidationNotice(
+                new FastTravelBetweenFarStopsNotice(
+                    tripTable.byTripId(stopsOnTrip.get(0).tripId()),
+                    stopsOnTrip.get(startIdx),
+                    startStop,
+                    stopsOnTrip.get(endIdx),
+                    endStop,
+                    speedKph,
+                    distanceToEndIdx));
+          }
+          // Report at most once for each trip.
+          return;
+        }
+      }
+    }
+  }
+
+  /**
+   * Validates travel speed between consecutive stops for all trips in a given line variant.
+   *
+   * <p>If there is a fast travel detected, then a separate notice is issued for each trip.
+   */
+  private void validateConsecutiveStops(
+      List<List<GtfsStopTime>> trips,
+      double[] distancesKm,
+      double maxSpeedInKph,
+      NoticeContainer noticeContainer) {
+    final List<GtfsStopTime> stopTimes = trips.get(0);
+
+    for (int i = 0; i < distancesKm.length; ++i) {
+      final GtfsStopTime stopTime1 = stopTimes.get(i);
+      final GtfsStopTime stopTime2 = stopTimes.get(i + 1);
+      if (!(stopTime1.hasDepartureTime() && stopTime2.hasArrivalTime())) {
+        continue;
+      }
+      final double distanceKm = distancesKm[i];
+      final double speedKph = getSpeedKphBetweenStops(distanceKm, stopTime1, stopTime2);
+      if (speedKph <= maxSpeedInKph) {
+        continue;
+      }
+      final GtfsStop stop1 = stopTable.byStopId(stopTime1.stopId());
+      final GtfsStop stop2 = stopTable.byStopId(stopTime2.stopId());
+      // Issue one notice per each trip.
+      for (List<GtfsStopTime> stopsOnTrip : trips) {
+        noticeContainer.addValidationNotice(
+            new FastTravelBetweenConsecutiveStopsNotice(
+                tripTable.byTripId(stopsOnTrip.get(0).tripId()),
+                stopsOnTrip.get(i),
+                stop1,
+                stopsOnTrip.get(i + 1),
+                stop2,
+                speedKph,
+                distanceKm));
+      }
+    }
+  }
+
+  static double getSpeedKphBetweenStops(
```

Comment:
Should this be private?

Reply:
StopTimeTravelSpeedValidatorTest uses getSpeedKphBetweenStops. Let me put `@VisibleForTesting` here.